src/server/api/schema.js

```diff
@@ -909,6 +923,28 @@ const rootMutations = {
 
       const replaceCurlyApostrophes = rawText => rawText.replace(/[\u2018\u2019]/g, "'")
 
+      let contactTimezone = {}
+      if (contact.timezone_offset) {
+        // couldn't look up the timezone by zip record, so we load it
+        // from the campaign_contact directly if it's there
+        const [offset, hasDST] = contact.timezone_offset.split('_')
+        contactTimezone.offset = parseInt(offset, 10)
+        contactTimezone.hasDST = hasDST === '1'
+      }
+
+      const sendBefore = getSendBeforeTimeUtc(
+        contactTimezone,
+        { textingHoursEnd: organization.texting_hours_end, textingHoursEnforced: organization.texting_hours_enforced },
+        {
+          textingHoursEnd: campaign.texting_hours_end,
+          overrideOrganizationTextingHours: campaign.override_organization_texting_hours,
+          textingHoursEnforced: campaign.texting_hours_enforced,
+          timezone: campaign.timezone
+        }
+      )
+
+      const sendBeforeDate = sendBefore ? sendBefore.toDate() : null
```

Comment:
if we are doing sendBefore here, shouldn't we also test whether we're already past sendBefore before creating the message?

-- i.e. if the API call is made past sendBefore we should error, right?

Reply:
👍